src/converters/shmup/mod.rs

```diff
@@ -104,11 +107,12 @@ fn blob_to_entity(
     blob: &extractor::git::Blob,
     files: &HashMap<String, extractor::code::FileData>,
 ) -> Entity {
+    let languages = languages::new();
     Entity {
         id: blob.sha.clone(),
         name: blob.name.clone(),
-        color: "a color".to_string(),
-        kind: "kind".to_string(),
+        color: get_color(blob, &languages),
+        kind: get_kind(blob, &languages),
```

Comment:
**Note:**
It seems to me that a significiant part of `get_color()` and `get_kind()` code is duplicated and thus unnecessarily executed twice.
Especially, in `languages::color_from_extension()` and `languages::kind_from_extension()` we browse the entire list of languages which is quite big, and I wonder if doing it twice rather than once doesn't have a significiant cost.

Reply:
That's a fair point the code in `languages::(color|kind)_from_extensions` are duplicated, mainly because the way we are getting colors is still kind of a hack and will diverge from the implementation of the way we get kinds. I decided to keep them separate until we find a cleaner way to determine the color or kind from an extension. The solution would be to replicate the code used by GitHub's linguist repository.

As for performance, that's something we should address in a separate PR and I think we can already feel the loss in performance as the converter seem way slower than the extractor and I think you have pointed out a place we could definitely speed things up here. I created an [issue](https://github.com/elhmn/waking-git/issues/53) to tackle this type of optimization.